### PR TITLE
Handoff ACPI and E820 to executable(PE/ELF) payload through HOB

### PR DESCRIPTION
**Pull request by gaojiaqi7**
Fix below issues:
https://github.com/confidential-containers/td-shim/issues/174
https://github.com/confidential-containers/td-shim/issues/94

### CODE REVIEW

File: td-payload/src/main.rs [hunk 1]
```diff
@@ -34,7 +44,95 @@ extern "win64" {
     fn cet_ss_test(count: usize);
 }
 
-#[cfg(not(test))]
+pub struct PayloadHobInfo<'a> {
+    hob_list: &'a [u8],
+    page_table: pi::hob::MemoryAllocationHeader,
+    payload: pi::hob::MemoryAllocationHeader,
+    runtime: pi::hob::MemoryAllocationHeader,
```
Comment (jyao1) [inline td-payload/src/main.rs:51]: Do we assume only one runtime memory block is available?

**Comment by jyao1** [comment]
please resolve conflict.

**Review by jyao1 — APPROVED** [review]

File: td-shim/src/lib.rs [hunk 2]
```diff
@@ -6,21 +6,55 @@
 
 use scroll::{Pread, Pwrite};
 
-use td_uefi_pi::pi;
+use td_uefi_pi::{pi, pi::guid};
```
Comment (jiangliu) [inline td-shim/src/lib.rs:9]: You may also use use td_uefi_pi::pi::{self, guid};

File: td-shim/src/lib.rs [hunk 3]
```diff
@@ -6,21 +6,55 @@
...
 pub mod secure_boot;
 
-pub const TD_ACPI_TABLE_HOB_GUID: [u8; 16] = [
-    0x70, 0x58, 0x0c, 0x6a, 0xed, 0xd4, 0xf4, 0x44, 0xa1, 0x35, 0xdd, 0x23, 0x8b, 0x6f, 0xc, 0x8d,
-];
-
-pub const TD_KERNEL_INFO_HOB_GUID: [u8; 16] = [
-    0x12, 0xa4, 0x6f, 0xb9, 0x1f, 0x46, 0xe3, 0x4b, 0x8c, 0xd, 0xad, 0x80, 0x5a, 0x49, 0x7a, 0xc0,
-];
+pub const TD_ACPI_TABLE_HOB_GUID: guid::Guid = guid::Guid::from_fields(
```
Comment (jiangliu) [inline td-shim/src/lib.rs:17]: Could you please help to add documentation for related sections in the td-shim spec?

File: td-payload/src/main.rs (hunk 1 above)
Comment (gaojiaqi7) [reply td-payload/src/main.rs:51]: The 'runtime' indicates the whole region located at the highest memory below 4G which contains heap, stack, evenlog and ACPI table etc., in runtime layout.

Do you think it is good to name this block as highmem or reserved?

Comment (jyao1) [reply td-payload/src/main.rs:51]: why we need one specific block to indicate highmem?

td-payload cannot make any assumption on td-shim, if it is not defined in td-shim spec.

**Review by jyao1 — CHANGES_REQUESTED** [review]
need remove 'runtime' from `PayloadHobInfo`

File: td-shim/src/lib.rs (hunk 2 above)
Comment (gaojiaqi7) [reply td-shim/src/lib.rs]: Thanks for reminding me

File: td-shim/src/lib.rs (hunk 3 above)
Comment (gaojiaqi7) [reply td-shim/src/lib.rs:21]: I added some references to the spec

**Comment by gaojiaqi7** [comment]
> need remove 'runtime' from `PayloadHobInfo`

hi @jyao1, 'runtime' has been removed and payload does not depend on interfaces that is not defined in td-shim spec.

**Comment by gaojiaqi7** [comment]
Move to main, close this one

**Pull request closed without merging**